orchestrator/relayer/src/fee_manager.rs

```diff
@@ -0,0 +1,219 @@
+                    Err(err) => {
+                        error!("Error while fetching token pair addresses {}", err);
+                        return false;
+                    }
+                    Ok(value) => value,
+                };
+
+                let token_addresses: HashMap<String, String> = match serde_json::from_str(&token_addresses_str)
+                {
+                    Err(err) => {
+                        error!("Error while parsing token pair addresses json configuration: {}", err);
+                        return false;
+                    }
+                    Ok(token_addresses) => token_addresses,
+                };
+
+                let api_url =
+                    std::env::var("TOKEN_API_URL").unwrap_or_else(|_| "https://cronos.org/gravity-testnet2/api/v1/oracle/quotes".to_owned());
+
+                for (key, value) in token_addresses.into_iter() {
+                    if value == "ETH" {
+                        self.token_api_url_map.insert(key, String::from("ETH"));
+                    } else {
+                        let request_url = format!("{url}/{pair}",
+                                                  url = api_url,
+                                                  pair = value);
+
+                        self.token_api_url_map.insert(key, request_url);
+                    }
+                }
+
+                return true;
+            }
+            RelayerMode::File => {
+                let config_file_path =
+                    std::env::var("TOKEN_PRICES_JSON").unwrap_or_else(|_| "token_prices.json".to_owned());
+
+                let config_str = match tokio::fs::read_to_string(config_file_path).await {
+                    Err(err) => {
+                        error!("Error while fetching token prices {}", err);
+                        return false;
+                    }
+                    Ok(value) => value,
+                };
+
+                let config: HashMap<String, String> = match serde_json::from_str(&config_str)
+                {
+                    Err(err) => {
+                        error!("Error while parsing token prices json configuration: {}", err);
+                        return false;
+                    }
+                    Ok(config) => config,
+                };
+
+                self.token_price_map = config;
+                return true;
+            }
+            _ => {
+                return true;
+            }
+        }
+    }
+
+    // A batch can be send either if
+    // - Mode is AlwaysRelay
+    // - Mode is either API or File and the batch has a profitable cost
+    // - Mode is either API or File and the batch has been waiting to be sent more than GRAVITY_BATCH_SENDING_SECS secs
+    pub async fn can_send_batch(
+        &mut self,
+        estimated_cost: &GasCost,
+        batch_fee: &Erc20Token,
+        contract_address: &EthAddress,
+    ) -> bool {
```

Comment:
I think `async` can be removed because there is no `async` function called in this function

Reply:
it is used in `get_token_price`